core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -911,4 +911,42 @@ class PlaintextConsumerTest extends BaseConsumerTest {
 
     assertThrows(classOf[WakeupException], () => consumer.position(topicPartition, Duration.ofSeconds(100)))
   }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersConsumerGroupProtocolOnly"))
+  def testCloseLeavesGroupOnInterrupt(quorum: String, groupProtocol: String): Unit = {
+    val adminClient = createAdminClient()
+    val consumer = createConsumer()
+    val groupId = consumerConfig.getProperty("group.id")
+
+    def hasMembers: Boolean = {
+      try {
+        val groupDescription = adminClient.describeConsumerGroups (Collections.singletonList (groupId) ).describedGroups.get (groupId).get
+        groupDescription.members.size() > 0
+      } catch {
+        case _: ExecutionException | _: InterruptedException =>
+          false
+      }
+    }
+
+    val listener = new TestConsumerReassignmentListener()
+    consumer.subscribe(List(topic).asJava, listener)
+    awaitRebalance(consumer, listener)
+
+    assertEquals(1, listener.callsToAssigned)
+    assertEquals(0, listener.callsToRevoked)
+    TestUtils.waitUntilTrue(() => hasMembers, s"Consumer did not join the consumer group within ${JTestUtils.DEFAULT_MAX_WAIT_MS} of subscribe")
+
+    try {
+      Thread.currentThread().interrupt()
+      assertThrows(classOf[InterruptException], () => consumer.close())
+    } finally {
+      // Clear the interrupted flag so we don't create problems for subsequent tests.
+      Thread.interrupted()
+    }
+
+    assertEquals(1, listener.callsToAssigned)
+    assertEquals(1, listener.callsToRevoked)
+    TestUtils.waitUntilTrue(() => !hasMembers, s"Consumer did not leave the consumer group within ${JTestUtils.DEFAULT_MAX_WAIT_MS} of interrupt/close")
```

Comment:
Won't this check always be true, regardless of the changes on this PR, the moment the session timeout expires? We should set a long session timeout, to be sure that the group ends up empty because the consumer left it intentionally even if interrupted (and not because its session expired and the broker kicked it out).

We can override the broker configs with the existing [brokerPropertiesOverrides](https://github.com/apache/kafka/blob/704476885ffb40cd3bf9b8f5c368c01eaee0a737/core/src/test/scala/integration/kafka/api/AbstractConsumerTest.scala#L70), and we should also be careful to do it only for this test (to not affect behaviour in others, like it's done [here](https://github.com/apache/kafka/blob/704476885ffb40cd3bf9b8f5c368c01eaee0a737/core/src/test/scala/integration/kafka/api/BaseAdminIntegrationTest.scala#L202) I believe).

Reply:
I explored the alternate approach you referred to with changing the configuration. It seemed to work, inasmuch as the tests pass locally for both `GroupProtocol` types. However, per the other comment about reproducibility locally vs. on CI, I'm not sure if I made any real difference.

However, I later updated the test to simply wait half of the configured session timeout, which I believe achieves the same effect. In this test, the session timeout is never explicitly set anywhere; it defaults to 45000 ms per the `ConsumerConfig` default. Waiting half of the session timeout means that the call to `waitUntilTrue` will fail before the broker gets around to kicking the consumer out of the group.

Correct me if I'm wrong, as I probably am because my AC is broken and my brain is sweating 😄

Reply:
Leaving the default session timeout and waiting for half of it definitely achieves the same result. The downside would be that waiting for half of it may not be enough for the group to become empty to the eyes of the admin client maybe? 

While playing with this when the issue first happened we confirmed that the classic consumer does send a leave group even if interrupted, that's why I still expect we should be able to have a test for it, and feels like we're close given that you confirm it consistently passes locally, but maybe just missing a tweak to have it passing in the CI?

Totally ok for me to leave it disabled for the classic for now and focus on the fix for the new consumer, but should we file a jira to enable it for the classic? (understand if it simply needs more time, or what's the reason for it to fail if the consumer sends the leave)

Reply:
Per above, I filed KAFKA-17397 to investigate the `ClassicKafkaConsumer` test failure.